test/test_datapipe.py

```diff
@@ -35,6 +41,63 @@
 
 T_co = TypeVar('T_co', covariant=True)
 
+PORT = 8000
```

Comment:
nit: Generate randomly or use tool to detect free one.

Reply:
fyi, now it uses port=0 or socketserver.TCPServer(("", 0), Handler) so it automatically picks an unused port.